crates/evm/core/src/backend/mod.rs

```diff
@@ -2016,28 +2022,79 @@ fn merge_zk_account_data<ExtDB: DatabaseRef>(
     fork_db: &mut ForkDB,
 ) {
     let mut merge_system_contract_entry = |system_contract: Address, slot: U256| {
-        let mut acc = if let Some(acc) = active.accounts.get(&system_contract).cloned() {
-            acc
-        } else {
-            // Account does not exist
-            return;
-        };
+        let Some(acc) = active.accounts.get(&system_contract) else { return };
+
+        // port contract cache over
+        if let Some(code) = active.contracts.get(&acc.info.code_hash) {
+            trace!("merging contract cache");
+            fork_db.contracts.insert(acc.info.code_hash, code.clone());
+        }
 
-        let mut storage = Map::<U256, U256>::default();
+        // prepare only the specified slot in account storage
+        let mut new_acc = acc.clone();
+        new_acc.storage = Default::default();
         if let Some(value) = acc.storage.get(&slot) {
-            storage.insert(slot, *value);
+            new_acc.storage.insert(slot, *value);
         }
 
-        if let Some(fork_account) = fork_db.accounts.get_mut(&system_contract) {
-            // This will merge the fork's tracked storage with active storage and update values
-            fork_account.storage.extend(storage);
-            // swap them so we can insert the account as whole in the next step
-            std::mem::swap(&mut fork_account.storage, &mut acc.storage);
-        } else {
-            std::mem::swap(&mut storage, &mut acc.storage)
+        // port account storage over
+        match fork_db.accounts.entry(system_contract) {
+            Entry::Vacant(vacant) => {
+                trace!("target account not present - inserting from active");
+                // if the fork_db doesn't have the target account
+                // insert the entire thing
+                vacant.insert(new_acc);
+            }
+            Entry::Occupied(mut occupied) => {
+                trace!("target account present - merging storage slots");
+                // if the fork_db does have the system,
+                // extend the existing storage (overriding)
+                let fork_account = occupied.get_mut();
+                fork_account.storage.extend(&new_acc.storage);
```

Comment:
Yes, it's because we only wish to migrate the slot belonging to the `addr` under a system contract. 
But this is the exact same functionality as before, only rewritten to be more expressive with the `entry`.